### PR TITLE
docs: document crate features

### LINKED ISSUE #2819: Document crate features
### Describe the feature

Some of our crates have features, some are specific to testing/debugging, others are optimizations etc., but they are largely undocumented.

Each crate that has features should have them listed out in their `lib.rs` file with a description of what the feature does, see e.g. https://docs.rs/tokio/latest/tokio/#feature-flags

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by onbjerg**
- Documents features of crates (closes #2819)
- Removes some features that were unused

### CODE REVIEW

File: crates/metrics/src/lib.rs [hunk 1]
```diff
@@ -6,6 +6,10 @@
 ))]
 
 //! Collection of metrics utilities.
+//! ## Feature Flags
```
Comment (onbjerg) [inline crates/metrics/src/lib.rs:9]:
```suggestion
//!
//! ## Feature Flags
```

**Comment by onbjerg** [comment]
@mattsse it appears the `reth-ipc` `client` feature does nothing, should this be the case? alternatively, we remove it

@joshieDo a lot of the crates have `bench` features, but they do nothing (no deps enabled, no `#[cfg]` in the code), is this intentional? otherwise I will remove

**Comment by joshieDo** [comment]
It's being used here:
https://github.com/paradigmxyz/reth/blob/main/crates/net/eth-wire/tests/fuzz_roundtrip.rs#L48
https://github.com/paradigmxyz/reth/blob/main/crates/storage/db/src/tables/codecs/fuzz/mod.rs#L15

My memory is a bit fuzzy, but it's necessary to being able to use them as cargo benchmarks across crates/dependencies

**Review by Rjected — APPROVED** [review]
nice!

**Review by mattsse — APPROVED** [review]
lgtm,

I think my intention was to feature gate the client impl.

I need to check if we need this at all

**Pull request merged**